pkg/storage/config/schema_config.go

```diff
@@ -282,6 +283,10 @@ func (cfg PeriodConfig) validate() error {
 		return validateError
 	}
 
+	if cfg.IndexType == TSDBType && cfg.IndexTables.Period != 24*time.Hour {
```

Comment:
Maybe you want to make [`tsdb.indexPeriod`](https://github.com/grafana/loki/pull/6273/files#diff-f74244a4225afefba7e0b4ba3c68b541d2d498a5fc0a0923f0c07a7d2e6e74cbR32) public and use it here:

```suggestion
	if cfg.IndexType == TSDBType && cfg.IndexTables.Period != tsdb.IndexPeriod {
```

Reply:
tried it already, it causes circular dependency since `tsdb` uses schema config :)

Reply:
taken care of this suggestion. Thanks for the review folks!